### PR TITLE
Make expert mode permanent even after manually switching it off and reconnecting

### LINKED ISSUE #2610: Option "Permanently activate expert mode" works only if BF-Configurator is restarted
New function on "Option "Permanently activate expert mode" Tab did not work perfect.

# BF-Configurator Version
image

image

# Problem
image
If switch is ON and user connect to FC "Enable expert mode"-Switch is ON - perfect.
If user switch manually to OFF "expert mode" and than disconnect from FC and reconnect to FC again ends with expert-Mode with OFF.

# Conclusion
Suggestion:  always read Option-states during connect and not only on start. This increase user experience

### PR DESCRIPTION
**Pull request by sobelek**
Hey.
This PR Fixes #2610
If you see any other possibility of implementing this please let me know and lets talk about it.

This is my first commit to this codebase so if I missed anything process related let me know.

### CODE REVIEW

File: src/js/serial_backend.js [hunk 1]
```diff
@@ -193,6 +193,12 @@ function finishClose(finishedCallback) {
     // reset active sensor indicators
     sensor_status(0);
 
+    // reset expert mode
+    ConfigStorage.get('permanentExpertMode', function (result) {
+        const checked = result.permanentExpertMode;
+        $('input[name="expertModeCheckbox"]').prop('checked', checked).change();
```
Comment (haslinghuis) [inline src/js/serial_backend.js:199]: We should use the new syntax:

```suggestion
        $('input[name="expertModeCheckbox"]').prop('checked', checked).trigger('change');
```

**Review by haslinghuis — CHANGES_REQUESTED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

File: src/js/serial_backend.js (hunk 1 above)
Comment (sobelek) [reply src/js/serial_backend.js]: Yea, didn't know that
Done :+1:

**Comment by mrRobot62** [comment]
tested on latest build - works as requested - Bug/Request fixed

**Review by haslinghuis — APPROVED** [review]

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> FAIL

**Review by KarateBrot — APPROVED** [review]

**Review by McGiverGim — APPROVED** [review]

File: src/js/serial_backend.js [hunk 2]
```diff
@@ -193,6 +193,12 @@ function finishClose(finishedCallback) {
     // reset active sensor indicators
     sensor_status(0);
 
+    // reset expert mode
+    ConfigStorage.get('permanentExpertMode', function (result) {
+        const checked = result.permanentExpertMode;
+        $('input[name="expertModeCheckbox"]').prop('checked', checked).trigger('change');
+    });
```
Comment (chmelevskij) [inline src/js/serial_backend.js:200]: There is not need to use callback syntax now (we should drop that completely...)
```suggestion
    const result = ConfigStorage.get('permanentExpertMode');
    const checked = result.permanentExpertMode;
    $('input[name="expertModeCheckbox"]').prop('checked', checked).trigger('change');
```

Comment (haslinghuis) [reply src/js/serial_backend.js:200]: @chmelevskij will be fixed in https://github.com/betaflight/betaflight-configurator/pull/2636

Comment (chmelevskij) [reply src/js/serial_backend.js:200]: Ah, though that one is in already.... soz

**Review by chmelevskij — APPROVED** [review]

File: src/js/serial_backend.js (hunk 2 above)
Comment (chmelevskij) [reply src/js/serial_backend.js:200]: But wait a minute, master already supports sync access, so wouldn't it be better to just go with the correct usage?

Comment (chmelevskij) [reply src/js/serial_backend.js:200]: https://github.com/betaflight/betaflight-configurator/blob/master/src/js/ConfigStorage.js#L32

**Comment by blckmn** [comment]
AUTOMERGE: (PASS)
 - github identifies PR as mergeable -> PASS
 - assigned to a milestone -> PASS
 - cooling off period lapsed -> PASS
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> PASS
 - `Tested` label found -> PASS
 - assigned to an approver -> PASS
 - approver count at least three -> PASS

**Comment by limonspb** [comment]
I think this one introduced the bug. On presets tab save/load backup buttons don't act normal. At the end they need to restart the configurator page and FC, but they don't anymore.

I can check what's happening only in a few days. Now I just tested nightly builds to see where the issue started: e62ab2d

**Pull request merged**